backend/src/migrations/20240221213010_add_user_achievements_table.ts

```diff
@@ -0,0 +1,40 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'user_achievements';
+
+const ColumnName = {
+    ID: 'id',
+    USER_ID: 'user_id',
+    ACHIEVEMENT_ID: 'achievement_id',
+    DATE_ACHIEVED: 'date_achieved',
+} as const;
+
+async function up(knex: Knex): Promise<void> {
+    return knex.schema.createTable(TABLE_NAME, (table) => {
```

Comment:
since you are extending AbstractModel it's expected that createdAt and updatedAt will be present.